process/sync/baseSync.go

```diff
@@ -197,6 +196,49 @@ func (boot *baseBootstrap) applyBlock(
 	return nil
 }
 
+func (boot *baseBootstrap) addHeadersToForkDetector(
+	shardId uint32,
+	nonce uint64,
+	getHeader func(uint32, uint64) (data.HeaderHandler, []byte, error)) {
```

Comment:
Also here, getHeader could be implemented by both bootstrappers and this can be changed to an interface.

Reply:
Done